src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -428,83 +450,74 @@ private async Task RefreshIndividualKeyValues(CancellationToken cancellationToke
 
                 bool hasChanged = false;
                 KeyValueIdentifier watchedKeyLabel = new KeyValueIdentifier(watchedKey, watchedLabel);
-                bool success = false;
 
-                try
+                if (_watchedSettings.TryGetValue(watchedKeyLabel, out ConfigurationSetting watchedKv))
                 {
-                    if (_watchedSettings.TryGetValue(watchedKeyLabel, out ConfigurationSetting watchedKv))
-                    {
-                        KeyValueChange keyValueChange = default;
-                        await TracingUtils.CallWithRequestTracing(_requestTracingEnabled, RequestType.Watch, _requestTracingOptions,
-                            async () => keyValueChange = await _client.GetKeyValueChange(watchedKv, cancellationToken).ConfigureAwait(false)).ConfigureAwait(false);
+                    KeyValueChange keyValueChange = default;
+                    await TracingUtils.CallWithRequestTracing(_requestTracingEnabled, RequestType.Watch, _requestTracingOptions,
+                        async () => keyValueChange = await client.GetKeyValueChange(watchedKv, cancellationToken).ConfigureAwait(false)).ConfigureAwait(false);

```

Comment:
Do we need `UpdateCacheExpirationTime(changeWatcher, success: true);` after this call succeeds?

Reply:
I updated the logic to only call `UpdateCacheExpirationTime` when the complete workflow (Refresh or Load) either succeeds or fails.